### ISSUE TITLE
Travis CI builds sometimes fail because jobs time out

### ISSUE THREAD

**Issue by BenjaminW3** [ComputationalRadiationPhysics/alpaka#414]
Mostly this is hit by job 16 and 19. ~These jobs natively compile CUDA using clang but have additionally OpenMP 4.0 enabled.
It could simply be the long test runtime due to the slow OpenMP 4.0 implementation. I am currently trying to add exact timing to the log output for better investigation.~

**Comment by psychocoderHPC** [comment]
For the failing test the setting `ALPAKA_CI_SANITIZERS=ASan+UBSan` is set. Can we create two tests, one with `ALPAKA_CI_SANITIZERS=ASan` and `ALPAKA_CI_SANITIZERS=UBSan` this should reduce the build time for the single test.

**Comment by BenjaminW3** [comment]
We can even remove `UBSan` from job 19 which has both `UBSan` and `ASan` because we have a seperate UBSan only job.
However, this seperate UBSan job 16 still times out sometimes. As far as I can see this is caused by this job never successfully uploading its docker image into the cache and always building it from scratch.
The cache upload seems to fail due to hard drive space limitations.

My timing experiments showed what I have expected. The test runtime is not the problem at all. Compiling is the bottleneck at least for job 19:
```
test/unit/atomic/src/AtomicTest.cpp ~1.5 min
test/unit/block/shared/src/BlockSharedMemDyn.cpp ~1.5 min
test/unit/block/shared/src/BlockSharedMemSt.cpp ~2.5 min
/test/unit/block/sync/src/BlockSync.cpp ~1.5 min
test/unit/block/sync/src/BlockSyncPredicate.cpp ~1.5 min
test/unit/kernel/src/KernelLambda.cpp ~3 min
test/unit/kernel/src/KernelWithAdditionalParam.cpp ~1.5 min
test/unit/kernel/src/KernelWithConstructorAndMember.cpp ~2 min
test/unit/kernel/src/KernelWithTemplate.cpp ~3 min
test/unit/mem/buf/src/BufTest.cpp ~3 min
test/unit/mem/view/src/ViewSubViewTest.cpp ~5 min
...
```

**Comment by BenjaminW3** [comment]
For now this should be solved with #433. Until the next time ;-)
